cloud-events-e2e-scenario/README.md

```diff
@@ -0,0 +1,182 @@
+* Name it `sample-external-solution`.
+
+## Establish secure connection between external solution and Kyma
+
+We will set up the secure connectivity between the mock external solution and the Kyma application `sample-external-solution`.
+
+Here we will use the [connector service](https://github.com/kyma-project/kyma/blob/master/docs/application-connector/docs/010-architecture-connector-service.md) to secure the communication.
+
+* Clone this [repository](https://github.com/janmedrek/one-click-integration-script).
+
+* Copy the token
+  * Navigate to the `Integration --> Applications --> sample-external-solution`
+  * `Connect Application`
+  * Copy the token to the clipboard
+  
+* Use the one-click-generation [helper script](https://github.com/janmedrek/one-click-integration-script) to generate the certificate
+
+  ```bash
+  ./one-click-integration.sh -u <paste token here>
+  ```
+
+  This will generate a `generated.pem` file. This contains the signed certificate and key that we will use for subsequent communications.
+
+  > **NOTE** The token is short-lived. So either hurry up or regenrate the token
+
+  ```bash
+  export NODE_PORT=$(kubectl get svc -n kyma-system application-connector-ingress-nginx-ingress-controller -o jsonpath='{.spec.ports[?(@.name=="https")].nodePort}')
+  ```
+
+## Mock External solution
+
+* Calls from external solution to kyma
+  * Use any http client e.g curl, httpie, postman etc.
+
+## Register Events
+
+* Register events via the metadata api using [register-events.json](./register-events.json)
+
+    ```bash
+    # Using httpie
+    http POST https://gateway.kyma.local/sample-external-solution/v1/metadata/services --cert=generated.pem --verify=no < register-events.json
+    # Using curl
+    curl -X POST -H "Content-Type: application/json" -d @./register-events.json https://gateway.kyma.local/sample-external-solution/v1/metadata/services --cert generated.pem -k
+    ```
+
+* Verify
+  * Navigate to Integration --> Applications --> sample-external-solution
+  * You should see the registered events.
+
+## Bind Application
+
+* Navigate to Integration --> Applications --> sample-external-solution
+* Create a binding with the Kyma namespace `workshop`.
+
+## Add events and APIs to the Kyma Namespace
+
+* Navigate to Service Catalog for Kyma namespace `workshop`
+
+* You should see the registered events available.
+* Click on the details
+* Add to your namespace
+* Check under `Service Instances` if it appears
+
+## Create Lambda
+
+* Create the lambda definition in `lambda ui` using [lambda.js](./lambda.js)
+  * Add label `app:<name-of-the-lambda>`
+* Select function trigger as event trigger, then choose `order.created`
+
+# Runtime
+
+## Publish the event
+
+* Using the generated certificate from the `./one-click-integration.sh`, send a Cloud Event to the gateway using any HTTP client e.g curl, httpie, postman etc. or a CloudEvents SDK 
+
+```go
+# Go Example
+package main
+
+import (
```

Comment:
The code in  the *.go file should be enough. No need to have it in the Readme too.